### ISSUE TITLE
Invalid serialization of quantity values with exponents

### ISSUE THREAD

**Issue by wetneb** [Wikidata/Wikidata-Toolkit#341]
Way to reproduce the issue:
```
BigDecimal amount = new BigDecimal("12.34E+5");
QuantityValue value = Datamodel.makeQuantityValue(amount, "1");
ObjectMapper mapper = new ObjectMapper();
System.out.println(mapper.writeValueAsString(value));
```
You get `{"value":{"amount":"+1.234E+6","unit":"1"},"type":"quantity"}`, which is refused by the Wikibase API. One format that works is `{"value":{"amount":"+1234000","unit":"1"},"type":"quantity"}`, but I am wondering if this is distorting the value input by the user (as we are losing the precision information that is implicitly encoded in the engineering notation). Is there a sensible way to derive lower and upper bounds and set them silently for the user?

In the meantime client code can work around the issue by converting their `BigDecimal` objects like this: `new BigDecimal(myBigDecimal.toPlainString())`.

**Comment by Tpt** [comment]
>  Is there a sensible way to derive lower and upper bounds and set them silently for the user?

I believe that "12.34E+5" may be encoded as "+1234000±1000". What do you think about it?

> In the meantime client code can work around the issue by converting their `BigDecimal` objects like this: `new BigDecimal(myBigDecimal.toPlainString())`.

`myBigDecimal.unscaledValue()` does the same thing and is even more convenient.

Before thinking about how to properly handle these values if they are given to the data model constructors we could just check if `bigDecimal.scale() == 0` and throw a meaningful error if if it not the case.

**Comment by wetneb** [comment]
I would say that it's rather "+1234000±500", no?

I'm confused - by rejecting everything that does not have scale 0, we would reject things like `12.345`, right?

**Comment by Tpt** [comment]
> I would say that it's rather "+1234000±500", no?

Yes, my bad. Engineering notation is indeed doing nearest-integer rounding.

>I'm confused - by rejecting everything that does not have scale 0, we would reject things like 12.345, right?

Yes, indeed. If we want to add a filter, we need to find an other way. Maybe `bigDec.toString().equals(bigDec.toPlainString())`? But it's not very nice/efficient. An other way to fix this issue would be to do not reject any value and call `toPlainString` everywhere.
